### PR TITLE
[List] Add Alternative Icon Support 

### PR DESCRIPTION
**Pull request by tomwanzek**
This PR:

* Adds an `iconChoice` property to the `List` component to allow use of an alternative icon for icon-styled list items
* Fix a minor styling issue with `AccordionItem` due to different border widths for normal and hover/focus states

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by tomwanzek** [comment]
@eliheron As discussed, as of right now, this PR still has an issue with the actual rendering of the `plusIcon`. I need a fresh set of 👓  on it 😄

File: packages/match-components/__tests__/list.test.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { axe } from "jest-axe";
 
 import { withTheme } from "@twilio-labs/match-themes";
 
-import { List, ListItem, ListVariant } from "../src";
+import { List, ListIconChoice, ListItem, ListVariant } from "../src";
```
Comment (eliheron) [inline packages/match-components/__tests__/list.test.tsx:7]: This is a really elegant solution to this. Great work!

**Review by eliheron — COMMENTED** [review]
🚀 The bulk of this pr looks good. I just couldn't get the svg to load

**Comment by tomwanzek** [comment]
@eliheron So the main glitch was an issue with the URL encoding of the fill color. It still contained a `#` for the hex, which was not encoded as `%23` 🤦 

I pushed a fix. However, I will still need to get a better SVG version from @serenabux . When I export it from Figma is comes in a 17px height instead of 16px, which has a minor clip effect at the bottom
Screen Shot 2021-12-01 at 11 27 40 AM
:

**Comment by tomwanzek** [comment]
"+" Icon fixed 🎉

**Review by eliheron — APPROVED** [review]
🚀

**Pull request merged**